### PR TITLE
Migrate Cart and Checkout Pages to the Template Editor when using a FSE theme

### LINKED ISSUE #9334: Page migration to blockified template
Part of Epic: Cart/Checkout/Order Received Templates https://github.com/woocommerce/woocommerce-blocks/issues/9288

For existing users using a block based theme, we need to migrate them to the block template away from the page. Rather than have a manual process, I propose:

- Seeing if we can populate the default template with the contents of the existing page so existing content is preserved
- Redirect the edit page link to the template editor
- ~See if we can drop the existing page in favour of endpoints or something similar~ _will be tackled separately_

### PR DESCRIPTION
**Pull request by mikejolley**
Adds a mechanism to migrate pages to the template editor by:

1. Taking default content for the template from the page
2. Redirecting the edit page link to the template

Default content is generated from the page content and the page title. This means if you're using a shortcode, or a block, on your cart/checkout page, it will automatically copy across to the FSE template.

Fixes #9334

### Testing

1. Ensure you're running a FSE theme and have cart/checkout pages already setup. Before using this, edit the page content to add something custom/unique e.g. some text, so you can identify the content later.
2. Check out this PR
3. Go to Appearance > Editor > Templates and view the Cart and Checkout templates. They should contain your original page content (unless already customised, in which case you can reset customisations to see the page content)
4. Go to Pages  and edit a Checkout or Cart page. You should be taken to the editor instead.

* [x] Do not include in the Testing Notes

### CODE REVIEW

File: src/Templates/CartTemplate.php [hunk 1]
```diff
@@ -13,8 +12,14 @@ class CartTemplate {
 
 	/**
 	 * Constructor.
+	 *
+	 * Templates require FSE theme support, so this will only init if a FSE theme is active.
```
Comment (nerrad) [inline src/Templates/CartTemplate.php:16]: Let's use the term "Block theme" as it is the "official term" now. WP community is moving away from "FSE theme" (or [FSE anything for that matter](https://make.wordpress.org/updates/2022/11/04/site-editor-a-more-user-friendly-name/))

**Review by nerrad — COMMENTED** [review]
While having the redirects is good to catch things like bookmarked items, or links added in ways we might forget, I think we should also consider filtering the page list table to remove the pages from any display.

That could be done in a followup though.

**Comment by mikejolley** [comment]
@nerrad one thing it does catch is the "edit page" link in the admin toolbar. So I agree it might be worth having both. Although...if we hide it, the user won't be able to remove that page manually. New users won't be affected if we find a way to prevent page creation on install.

**Comment by nerrad** [comment]
> Although...if we hide it, the user won't be able to remove that page manually.

Well, I think we need to keep the page in the short term in case the user switches back to a classic theme... which is also a scenario we might need to account for in terms of behaviour. 

The primary value I see for hiding the page from the list table is to reduce confusion when people visit there as the entry point to view/edit their pages. While the redirect mitigates that confusion, the quick edit functionality is still there.

**Review by wavvves — APPROVED** [review]
Just wow, this is great 🤩

**Pull request merged**